lib/helpers/preset-announcement.js

```diff
@@ -2,36 +2,45 @@
 const logger = require('sonos-discovery/lib/helpers/logger');
 const isRadioOrLineIn = require('../helpers/is-radio-or-line-in');
 
-function saveAll(system) {
+function savePresetPlayers(system, pset) {
   const backupPresets = system.zones.map((zone) => {
     const coordinator = zone.coordinator;
     const state = coordinator.state;
-    const preset = {
-      players: [
-        { roomName: coordinator.roomName, volume: state.volume }
-      ],
-      state: state.playbackState,
-      uri: coordinator.avTransportUri,
-      metadata: coordinator.avTransportUriMetadata,
-      playMode: {
-        repeat: state.playMode.repeat
-      }
-    };
-
-    if (!isRadioOrLineIn(preset.uri)) {
-      preset.trackNo = state.trackNo;
-      preset.elapsedTime = state.elapsedTime;
-    }
+    for (var index = 0; index < pset.players.length; ++index) {
+			if (pset.players[index].roomName == coordinator.roomName) {
```

Comment:
I'm not really following the logic here, why would you create a preset for each player in a zone?

Reply:
The point is to completely ignore other players from my house. When my daughter was listening spotify in her room, if I had a preset announcement in any other room(s) , at the moment of tts ending, her player track got rewind at the moment of preset backup. Her player was not supposed to be restored at all.
I couldn't find a way to restore only specific player from an allSave preset, so I decided to create separate presets then restore all available.